workbench/test/test_filethumbs.py

```diff
@@ -55,6 +56,7 @@ def test_three_thumbs_initial_state(self):
             initial_up = int(up_count.text)
             initial_down = int(down_count.text)
             thumb.find_element_by_css_selector('span.upvote').click()
+            time.sleep(5)  # this is a dirty, dirty hack. Feel free to remove if you're working in this code.
```

Comment:
This test class violates most of the recommendations in http://bok-choy.readthedocs.io/en/latest/guidelines.html#test-design-guidelines .  The quick fix is to use an `EmptyPromise` to wait for the expected outcome, as shown in one of the examples in that doc.  The more complete fix is to create a PageObject subclass which bundles all the direct browser interactions into logical groupings and use that to execute the tests (as also explained in that doc).  Then you'd have access to convenience methods like [wait_for()](https://github.com/edx/bok-choy/blob/master/bok_choy/page_object.py#L550).